drivers/flash/soc_flash_nrf_ticker.c

```diff
@@ -35,20 +35,36 @@ struct ticker_sync_context {
 
 static struct ticker_sync_context _ticker_sync_context;
 
-
 /* semaphore for synchronization of flash operations */
 static struct k_sem sem_sync;
 
-static inline int _ticker_stop(uint8_t inst_idx, uint8_t u_id, uint8_t tic_id)
+static void ticker_stop_work_cb(uint32_t status, void *param)
 {
-	int ret = ticker_stop(inst_idx, u_id, tic_id, NULL, NULL);
+	__ASSERT(status == TICKER_STATUS_SUCCESS,
+		 "Failed to stop work ticker.\n");
```

Comment:
This assert is invalid. When not in race condition, i.e. prepare ticker was stopped with no outstanding work ticker scheduled, then `status` will be `TICKER_STATUS_FAILURE`.
```suggestion
	__ASSERT((status == TICKER_STATUS_SUCCESS || status == TICKER_STATUS_FAILURE),
		 "Failed due to work ticker stop resulted in ticker job busy.\n");
```